deb/ubuntu-cosmic/Dockerfile.aarch64

```diff
@@ -0,0 +1,29 @@
+FROM ubuntu:cosmic
+
+RUN apt-get update && apt-get install -y apparmor bash-completion libbtrfs-dev build-essential cmake curl ca-certificates debhelper dh-apparmor dh-systemd git libapparmor-dev libdevmapper-dev libltdl-dev libseccomp-dev pkg-config vim-common libsystemd-dev --no-install-recommends && rm -rf /var/lib/apt/lists/*
+
+ARG GO_VERSION
+
+RUN curl -fSL "https://golang.org/dl/go${GO_VERSION}.linux-arm64.tar.gz" | tar xzC /usr/local
+ENV GOPATH /go
+ENV PATH $PATH:/usr/local/go/bin:$GOPATH/bin
+ENV DOCKER_BUILDTAGS apparmor seccomp selinux
+ENV RUNC_BUILDTAGS apparmor seccomp selinux
+
+COPY common/ /root/build-deb/debian
+COPY build-deb /root/build-deb/build-deb
+
+RUN mkdir -p /go/src/github.com/docker && \
```

Comment:
Not sure how important caching is for these Dockerfiles, but this could be done earlier in the Dockerfile.

There's some other lines as well that could be put earlier, to prevent churn / unneeded re-creation of layers

Reply:
(very) quick attempt to re-order a bit more optimised (but possibly this should be done for all Dockerfiles afterwards (haven't checked);

```Dockerfile
FROM ubuntu:cosmic

ENV DISTRO ubuntu
ENV SUITE cosmic

ENV GOPATH /go
ENV PATH $PATH:/usr/local/go/bin:$GOPATH/bin
ENV DOCKER_BUILDTAGS apparmor seccomp selinux
ENV RUNC_BUILDTAGS apparmor seccomp selinux

RUN apt-get update
	&& apt-get install -y \
		apparmor \
		bash-completion \
		build-essential \
		ca-certificates \
		cmake curl \
		debhelper \
		dh-apparmor \
		dh-systemd \
		git \
		libapparmor-dev \
		libbtrfs-dev \
		libdevmapper-dev \
		libltdl-dev \
		libseccomp-dev \
		libsystemd-dev \
		pkg-config \
		vim-common \
	--no-install-recommends \
	&& rm -rf /var/lib/apt/lists/*

RUN mkdir -p /go/src/github.com/docker && \
	mkdir -p /go/src/github.com/opencontainers && \
	ln -snf /engine /root/build-deb/engine && \
	ln -snf /cli /root/build-deb/cli && \
	ln -snf /root/build-deb/engine /go/src/github.com/docker/docker && \
	ln -snf /root/build-deb/cli /go/src/github.com/docker/cli

WORKDIR /root/build-deb
ENTRYPOINT ["/root/build-deb/build-deb"]

ARG GO_VERSION

RUN curl -fSL "https://golang.org/dl/go${GO_VERSION}.linux-arm64.tar.gz" | tar xzC /usr/local

COPY common/ /root/build-deb/debian
COPY build-deb /root/build-deb/build-deb
```